src/visualization/adapt.jl

```diff
@@ -0,0 +1,58 @@
+"""
+    adapt_to_mesh_level!(u_ode::AbstractVector, semi, level)
+    adapt_to_mesh_level!(sol::Union{DiffEqBase.ODESolution,TimeIntegratorSolution}, level)
+
+Use the regular adaptive mesh refinement routines to adaptively refine/coarsen the solution `u_ode`
+with semidiscretization `semi` towards a uniformly refined grid with refinement level `level`. The
+solution and parts of the semidiscretization (mesh and caches) will be *modified in place*.
+
+A convenience method accepts an ODE solution object, from which solution and semidiscretization are
+extracted as needed.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+
+See also: [`adapt_to_mesh_level`](@ref)
```

Comment:
Do you really want to duplicate the docstring? I would prefer using the same approach as base Julia, e.g. `map`/`map!`. Thus, 
```suggestion
Like [`adapt_to_mesh_level`](@ref), but modifies the solution and parts of the
semidiscretization (mesh and caches) in place.
```
Otherwise, we need to keep duplicated parts in sync... But you may convince me if you have a good argument :wink:

Reply:
Excellent point!